src/dune_rules/pkg_rules.ml

```diff
@@ -1819,12 +2127,12 @@ let find_package ctx pkg =
   | false -> Memo.return None
   | true ->
     let* db = DB.get ctx in
-    Resolve.resolve db ctx (Loc.none, pkg)
-    >>| (function
-           | `System_provided -> Action_builder.return ()
-           | `Inside_lock_dir pkg ->
-             let open Action_builder.O in
-             let+ _cookie = (Pkg_installed.of_paths pkg.paths).cookie in
-             ())
-    >>| Option.some
+    let+ pkg = Resolve.resolve db ctx (Loc.none, pkg) in
+    Some
+      (if pkg.system_provided
```

Comment:
Again this change seems gratuitous. With the old code, the return type forced callers to handle "system provided" packages, now one has to remember to look up some field.